src/databases/functions/users/user-unblock-by-admin.js

```diff
@@ -0,0 +1,7 @@
+const {User} = require('../../models');
+
+module.exports = async (userId) => {
+  if (userId == null) return null;
+  const result = await User.update({blocked_by_admin: false}, {where: {user_id: userId}});
+  return result;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a potential issue with the loose equality check on line 4. It's generally better to use strict equality (`===`) in JavaScript to avoid unexpected type coercion. Also, it would be beneficial to add error handling for the database operation.

```diff
- if (userId == null) return null;
+ if (userId === null || userId === undefined) return null;

- const result = await User.update({blocked_by_admin: false}, {where: {user_id: userId}});
+ let result;
+ try {
+   result = await User.update({blocked_by_admin: false}, {where: {user_id: userId}});
+ } catch (error) {
+   console.error(`Failed to unblock user with ID ${userId}:`, error);
+   throw error;
+ }
```
This change ensures that both `null` and `undefined` values for `userId` are handled correctly, and any errors during the update operation are logged and re-thrown.


<!-- This is an auto-generated comment by OSS CodeRabbit -->